src/inmanta/data/__init__.py

```diff
@@ -2694,7 +2693,11 @@ async def clear(self, connection: Optional[asyncpg.connection.Connection] = None
             await Compile.delete_all(environment=self.id, connection=con)  # Triggers cascading delete on report table
             await Parameter.delete_all(environment=self.id, connection=con)
             await Notification.delete_all(environment=self.id, connection=con)
-            await Code.delete_all(environment=self.id, connection=con)
+
+            await self._execute_query("DELETE FROM public.module WHERE environment=$1", self.id, connection=con)
+            await self._execute_query("DELETE FROM public.files_in_module WHERE environment=$1", self.id, connection=con)
+            await self._execute_query("DELETE FROM public.modules_for_agent WHERE environment=$1", self.id, connection=con)
```

Comment:
In the past, we would place these in methods on the relevant type, this is now in-line.
Could you motivate this change?

Reply:
Factorized this in the sqlalchemy base class.